dev/gen_apidocs/Dockerfile

```diff
@@ -0,0 +1,33 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM continuumio/anaconda3
+ADD . /apache-arrow
+WORKDIR /apache-arrow
+RUN apt-get update; apt-get install -y pkg-config g++ gcc libgtk2.0-dev \
+    libjemalloc-dev libboost-dev make autogen gtk-doc-tools \
+    gobject-introspection git \
+    libgirepository1.0-dev autoconf-archive \
+    libboost-filesystem-dev libboost-system-dev; \
+    conda install -c conda-forge doxygen maven cmake libthrift; \
+    conda create -y -q -n pyarrow-dev \
+        python=3.6 numpy six setuptools cython pandas pytest \
+        cmake flatbuffers rapidjson boost-cpp thrift-cpp snappy zlib \
+        brotli jemalloc lz4-c zstd ipython matplotlib numpydoc \
+        sphinx sphinx_bootstrap_theme \
+        -c conda-forge
```

Comment:
To avoid special snowflakes, I might prefer separating the C++ toolchain from the Python toolchain like we do in the CI scripts. Here they are commingled

Reply:
Done